src/uct/ib/base/ib_md.c

```diff
@@ -1295,7 +702,7 @@ static void uct_ib_md_release_device_config(uct_ib_md_t *md)
 }
 
 static ucs_status_t
-uct_ib_md_parse_reg_methods(uct_ib_md_t *md, uct_md_attr_t *md_attr,
+uct_ib_md_parse_reg_methods(uct_ib_md_t *md, uct_ib_md_attr_t *md_attr,
```

Comment:
1. not inherit uct_ib_md_attr_t  from uct_md_attr_t . rename to uct_ib_md_init_attr_t 
2. do not pass uct_ib_md_attr_t ; instead, pass `int rcache_ucm_events` in uct_ib_md_init_attr_t